caffe2/sgd/adagrad_op.h

```diff
@@ -32,12 +32,12 @@ void adagrad_compute(
     float* nw,
     float* nh,
     float epsilon,
-    float lr,
+    const float* lr,
```

Comment:
Just because `RunOnDevice` calls `adagrad_compute` for both CPU and CUDA context. In case of CUDA context, we can't dereference the device pointer on the host. So we need to either send a pointer to `adagrad_compute` or first do a `cudaMemcpy` to copy the content of LR to the host. I picked the first option because `cudaMemcpy` is expensive.